kafkaesk/consumer.py

```diff
@@ -382,15 +389,17 @@ async def _maybe_commit(self, forced: bool = False) -> None:
         if not self._auto_commit:
             return
 
-        if not self._consumer.assignment:
+        if not self._consumer.assignment or not self._tp:
             logger.warning("Cannot commit because no partitions are assigned!")
             return
 
         interval = self._app.kafka_settings.get("auto_commit_interval_ms", 2000) / 1000
         now = time.monotonic_ns()
         if forced or (now > (self._last_commit + interval)):
             try:
-                await self._consumer.commit()
+                offsets = {tp: offset for tp, offset in self._tp.items() if offset}
```

Comment:
Maybe we shouldn't store `None` offsets?

Reply:
yeah, i agree.